rules/CodeQuality/NodeAnalyser/AssertMethodAnalyzer.php

```diff
@@ -36,26 +37,47 @@ public function detectTestCaseCall(MethodCall|StaticCall $call): bool
         $methodName = $this->nodeNameResolver->getName($call->name);
         if (! str_starts_with((string) $methodName, 'assert') && ! in_array(
             $methodName,
-            NonAssertStaticableMethods::ALL
+            NonAssertPrefixedAssertionMethods::ALL
```

Comment:
this may need to check $call is instanceof StaticCall

Reply:
Can you explain a bit more why it is needed?

I renamed `NonAssertStaticableMethods` to `NonAssertPrefixedAssertionMethods` exactly because these methods are a mix of static and non-static methods, and the previous name was misleading

Reply:
The `$call` can be `MethodCall` or `StaticCall`, so only check when it is `StaticCall`

Reply:
I don't understand the full motivation behind this, but I committed it.

Thank you for the guidance.